microservice/microservice.go

```diff
@@ -150,32 +151,100 @@ func MicroserviceReadyForUpgrade(msdef *persistence.MicroserviceDefinition, db *
 	return true
 }
 
-// check if the given msdef needs rollback
-func MicroserviceNeedsRollback(msdef *persistence.MicroserviceDefinition) bool {
-	glog.V(5).Infof("Check if microservice needs to rollback: %v.", msdef)
+// Get the version range that the microservice should be in. Assuming there is one microservice version on the node at a time.
+// This function, for the pattern case, get a list of microservice references from all the workloads, it then gets the intersection
+// of the version ranges. For the non-pattern case, it gets the version range from the user input from the /servcie api.
+func GetMicroserviceUpgradeVersionRange(getWorkload exchange.WorkloadHandler,
```

Comment:
This looks a lot like the code in the microservice config API, can we find a way to make a reusable function that both pieces of code can use?

Reply:
This function is currently not used after the discussion we had that we were not going to handle the case when workloads change and hence microservice version ranges are changing.  This function gets the common version range for one microservice. The one in the config API needs version ranges for all the microservices and it does other things in the loop too.  We can remove this function for now, or keep it for future use. What is your suggestion?

Reply:
Remove it please. I dont plan to have a long future with this code.